src/vzlogger.cpp

```diff
@@ -105,6 +105,46 @@ const char *long_options_descs[] = {
 	NULL /* stop condition for iterator */
 };
 
+void openLogfile() {
+	FILE *logfd = fopen(options.log().c_str(), "a");
```

Comment:
fopen and most likely some more functions are not async-signal-safe: see https://man7.org/linux/man-pages/man7/signal-safety.7.html.

You could add a workaround by setting a flag/boolean only from the signal handler and doing the real processing later on from any other thread.

Reply:
pushed a fix
(separate commit for clarity, will squash that before merge)